mixer/pkg/protobuf/yaml/dynamic/handler.go

```diff
@@ -135,11 +141,18 @@ func (h *Handler) connect() (err error) {
 	return nil
 }
 
-func (h *Handler) handleRemote(ctx context.Context, instanceName string, qr proto.Marshaler,
-	dedupID string, resultPtr interface{}, encodedInstances ...[]byte) error {
-	svc := h.svcMap[instanceName]
+func (h *Handler) handleRemote(ctx context.Context, qr proto.Marshaler,
+	dedupID string, resultPtr interface{}, encodedInstances ...*adapter.EncodedInstance) error {
+	svc := h.svcMap[encodedInstances[0].Name]
```

Comment:
any concerns of IOOB

Reply:
handleRemote is an internal function. It is always called with at least one encodedInstance.